src/overlays/actors/ovl_En_Ice_Hono/z_en_ice_hono.c

```diff
@@ -21,37 +32,345 @@ const ActorInit En_Ice_Hono_InitVars = {
+    EnIceHono* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChainCapturableFlame);
+    Actor_SetScale(&this->actor, 0.0074f);
+    this->actor.flags |= 1;
+    Actor_SetHeight(&this->actor, 10.0f);
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInitCapturableFlame);
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+
+    this->actor.colChkInfo.mass = 0xFF;
+    EnIceHono_SetupActionCapturableFlame(this);
+}
+
+void EnIceHono_InitDroppedFlame(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChainDroppedFlame);
+    this->actor.scale.x = this->actor.scale.z = this->actor.scale.y = 0.00002f;
+    this->actor.gravity = -0.3f;
+    this->actor.minVelocityY = -4.0f;
+    this->actor.shape.unk_08 = 0.0f;
+    this->actor.shape.rot.x = this->actor.shape.rot.y = this->actor.shape.rot.z = this->actor.posRot.rot.x =
+        this->actor.posRot.rot.y = this->actor.posRot.rot.z = 0;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInitDroppedFlame);
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+
+    this->collider.dim.radius = this->actor.scale.x * 4000.4f;
+    this->collider.dim.height = this->actor.scale.y * 8000.2f;
+    this->actor.colChkInfo.mass = 0xFD;
+    EnIceHono_SetupActionDroppedFlame(this);
+}
+
+void EnIceHono_InitSmallFlame(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChainSmallFlame);
+    this->actor.scale.x = this->actor.scale.z = this->actor.scale.y = 0.0008f;
+    this->actor.gravity = -0.3f;
+    this->actor.minVelocityY = -4.0f;
+    this->actor.shape.unk_08 = 0.0f;
+
+    EnIceHono_SetupActionSmallFlame(this);
+}
+
+void EnIceHono_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+    s16 params = this->actor.params;
+
+    switch (this->actor.params) {
+        case -1:
+            EnIceHono_InitCapturableFlame(&this->actor, globalCtx);
+            break;
+        case 0:
+            EnIceHono_InitDroppedFlame(&this->actor, globalCtx);
+            break;
+        case 1:
+        case 2:
+            EnIceHono_InitSmallFlame(&this->actor, globalCtx);
+            break;
+    }
+
+    if ((this->actor.params == -1) || (this->actor.params == 0)) {
+        Lights_PointNoGlowSetInfo(&this->lightInfo, this->actor.posRot.pos.x,
+                                        (s16)this->actor.posRot.pos.y + 0x0A, this->actor.posRot.pos.z, 0x9B, 0xD2,
+                                        0xFF, 0);
+        this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+        this->unk154 = Math_Rand_ZeroOne() * 65535.5f;
+        this->unk156 = Math_Rand_ZeroOne() * 65535.5f;
```

Comment:
`65535.5f = 0x1FFFF/2.0f` which may be better since those are used as angles? (in sin/cos)